### DISCUSSION TITLE
Quarkus + Hibernate + Trino + Cassandra Connector

### DISCUSSION THREAD

**Discussion by jaiprasad266** [quarkusio/quarkus#25164] [Q&A]
Hi ,

We are trying to fetch data from trino for the cassandra connector using the quarkus rest endpoint , facing multiple exceptions like dialect not available .

Can anyone help us in using quarkus + hibernate with trino .

Please comment
<img width="558" alt="cassandra_trino_config" src="https://user-images.githubusercontent.com/54102261/165267039-f661cc84-26b6-439f-aabb-31d9d2c5e7e7.PNG">
.

[a comment by an automated account is not shown]

**Comment by Sanne** [comment]
hi @jaiprasad266 ,
that looks interesting but we don't have support for the Trino JDBC driver at the moment. 

Somebody should help contributing support for it; but I'm not sure how complex that could be: a requirement we have is that all components need to also work in graalvm native-images.

I see it's open source - that helps; still it looks like a fairly large, complex project compared to other JDBC drivers.

We also have basic support for "generic JDBC" drivers: this won't be optimally efficient as with a proper extension and is unlikely to work out of the box in native, but you should be able to configure it by referring to this section of the docs:
 - https://quarkus.io/guides/datasource#other-databases

This line is particularly important:
```
quarkus.datasource.db-kind=other
```
You'll also need to configure a Dialect explicitly; I don't know if there is an Hibernate ORM dialect for Trino.

That should work fine in JVM mode. Might be interesting to try native-image as well, but in that case I would expect it to need some additional metadata; you wouldn't have that need if there was a proper extension.

**Reply by JayGhiya** [reply]
@Sanne -> We are looking at adopting quarkus as part of our entire backend. One of core functionality is we using distributed query for multiple usecases including analytics. So let us say we decide to contribute to write a extension for quarkus - we would need a prerequisite checklist that future contributors have to go through to do this. like understanding vertx, panache fundamentals etc.. if you have that in place we can review and start.

**Reply by Sanne** [reply]
There is this section of the documentation that might be useful:
 - https://quarkus.io/guides/#writing-extensions

When creating a JDBC driver extension, it might also be useful to look at the soruce code of the other extensions for jdbc drivers we already have:
 - https://github.com/quarkusio/quarkus/tree/main/extensions/jdbc

**Reply by gsmet** [reply]
I think first you should understand exactly what the problem is. Because you were stating `dialect not available` which would say that Hibernate ORM doesn't have a dialect for Trino (which is kinda expected) and in this case, it's more on the ORM side that the work should be done (but I would expect the ORM team to say the dialect should be maintained outside of the ORM codebase).

Not having a JDBC Quarkus extension for Trino is not blocking per se if you have a dialect: you just have to define the dialect in the config yourself with `quarkus.hibernate-orm.dialect`.

**Reply by JayGhiya** [reply]
Got it so makes sense. i also referred https://trinodb.slack.com/archives/CFLB9AMBN/p1585224657034200 and it talks about why it is difficult to implement trino dialect for hibernate due to some non-oltp things that presto does. Also trino promises to be olap not oltp. so yea we can close the issue if this makes sense.

**Comment by JayGhiya** [comment]
We will try to come up with a open source example in terms of how to work with trino in a optimal way. that should help the audience if they want to use trino with quarkus
